crates/rolldown/src/dev/build_driver.rs

```diff
@@ -92,10 +92,40 @@ impl BuildDriver {
     }
   }
 
-  pub async fn ensure_latest_build(&self) -> BuildResult<()> {
-    if let Some((future, _)) = self.schedule_build_if_stale().await? {
-      future.await;
+  pub async fn ensure_latest_build_output(&self) -> BuildResult<()> {
+    let mut count = 0;
+
+    loop {
```

Comment:
When is this `loop` needed?

Reply:
When there's two queued hmr task. You need wait for them to finish.

Reply:
Ah, makes sense.